src/axom/mir/docs/sphinx/mir_utilities.rst

```diff
@@ -0,0 +1,235 @@
+.. ## Copyright (c) 2017-2024, Lawrence Livermore National Security, LLC and
+.. ## other Axom Project Developers. See the top-level LICENSE file for details.
+.. ##
+.. ## SPDX-License-Identifier: (BSD-3-Clause)
+
+******************************************************
+MIR Blueprint Utilities
+******************************************************
+
+The MIR component contains several useful building blocks for writing algorithms
+for Blueprint meshes.
+
+ * Structured as classes with an ``execute()`` method
+ * Often templated on execution space and views
+
+#######################
+Copying Blueprint Data
+#######################
+
+If a ``conduit::Node`` containing Blueprint data is not on the desired memory space, it can be
+moved using the ``axom::mir::utilities::blueprint::copy<ExecSpace>()`` function. The ``copy``
+function copies the source ``conduit::Node`` to the destination ``conduit::Node``, making sure
+to use the appropriate Axom allocator for non-string bulk arrays (e.g. arrays of ints, floats,
+doubles, etc.). Data small enough to fit in a ``conduit::Node`` and strings are left in the host
+memory space, which lets algorithms on the host side query them. For data that have been moved
+to the device, their sizes and data types can still be queried using normal Conduit mechanisms
+such as the ``conduit::Node::dtype()`` method.
+
+.. code-block:: cpp
+
+    conduit::Node hostMesh, deviceMesh, hostMesh2;
+    // host->device
+    axom::mir::utilities::blueprint::copy<axom::HIP_EXEC<256>>(deviceMesh, hostMesh);
+    // device->host
+    axom::mir::utilities::blueprint::copy<axom::SEQ_EXEC>(hostMesh2, deviceMesh);
+
+############################
+ConduitAllocateThroughAxom
+############################
+
+When writing algorithms that construct Blueprint data, it is helpful to force Conduit
+to allocate its memory through Axom's allocation routines and then make an ``axom::ArrayView``
+of the data in the Conduit node. This prevents data from having to be copied from an Axom
+data structure into a Conduit node since it can be constructed from the start inside the
+Conduit node. The size of the array must be known.
+
+The ``axom::mir::utilities::blueprint::ConduitAllocateThroughAxom``
+class is a template class that takes an execution space as a template argument and it
```

Comment:
break sentence:  "...as a template argument.  It installs an allocation routine..."

Reply:
Done